nautobot_floor_plan/tests/test_models.py

```diff
@@ -134,6 +137,26 @@ def test_create_floor_plan_tile_invalid_duplicate_rack(self):
                 floor_plan=self.floor_plans[2], x_origin=2, y_origin=2, status=self.active_status, rack=self.rack
             ).validated_save()
 
+    def test_create_floor_plan_tile_invalid_rack_rackgroup(self):
+        """A Rack being placed on a Rackgroup tile must also be in the reck_group."""
+        models.FloorPlanTile(
+            floor_plan=self.floor_plans[2],
+            x_origin=1,
+            y_origin=1,
+            status=self.active_status,
+            rack_group=self.rack_group,
+        ).validated_save()
+        # How about a rack without the correct rack group?
+        non_rack_group_rack = Rack.objects.create(name="Rack 2", status=self.active_status, location=self.floors[2])
+        with self.assertRaises(ValidationError):
+            models.FloorPlanTile(
+                floor_plan=self.floor_plans[2],
+                x_origin=1,
+                y_origin=1,
+                status=self.active_status,
+                rack=non_rack_group_rack,
+            ).validated_save()
```

Comment:
Would be good to see test coverage for each of the new checks in `group_tile_bounds()`.